### PR TITLE
fix(agents-core): make StreamedRunResult.currentTurn reflect the real turn count

### PR DESCRIPTION
**Pull request by dkindlund**
## What

`StreamedRunResult.currentTurn` is declared as a plain field initialized to `0` and never updated, so it always reads `0` regardless of how many turns a streamed run executes. Only the private `RunState._currentTurn` is incremented (once per model request in `beginTurn`).

This makes `currentTurn` unusable for callers that budget or report turns on a streamed run — for example re-invoking `run(agent, history, { stream: true, maxTurns })` between nudges and decrementing `maxTurns` by the turns already consumed. The field never moves off `0`.

## Fix

Convert `currentTurn` to a getter that delegates to `this.state._currentTurn`, matching `currentAgent` (the accessor immediately above it) and the other state-delegating getters on `RunResultBase`. Nothing in the package ever assigns `.currentTurn`, so a getter-only is safe.

```diff
-  public currentTurn: number = 0;
+  public get currentTurn(): number {
+    return this.state._currentTurn;
+  }
```

## Test

Adds a regression test in `packages/agents-core/test/result.test.ts` asserting `currentTurn` tracks `RunState._currentTurn` (0 initially, then reflects an incremented value). Verified it **fails on the current code** (`expected +0 to be 3`) and **passes with the fix**. `vitest run test/result.test.ts` (18 passed) and `tsc --noEmit` are green; changeset included.

## Notes

Latest published `@openai/agents-core` (0.13.4) and `main` both still have the dead field. Happy to adjust if you'd rather keep `currentTurn` a plain field like `maxTurns` and set it from the runner instead — the getter felt most consistent with the surrounding accessors.

### CODE REVIEW

**Review by seratch — CHANGES_REQUESTED** [review]
Before we can merge this, please make the public currentTurn value represent the number of turns actually admitted by the configured limit rather than exposing _currentTurn verbatim. beginTurn increments the state counter before checking maxTurns, so maxTurns: 0 with a handled max-turn error currently completes without a model request but would report currentTurn === 1. Please keep the public value capped at maxTurns and add streaming-run regression tests covering both a normal multi-turn run and the handled max-turn boundary; the current test only mutates the private counter directly.

Please also update the changeset summary to use the required `fix: ...` format. Once those focused changes are covered, this should be ready for another review.

**Comment by dkindlund** [comment]
@seratch Thanks for the review! Rebased onto latest `main` and pushed a follow-up that:

- **Caps `currentTurn` at `maxTurns`** — `max === null ? current : Math.min(current, max)`. Since `_currentTurn` is incremented before the limit check in `turnPreparation`, a handled `maxTurns: 0` boundary would otherwise report `1` for a run that admitted no turns; uncapped when `maxTurns` is `null`.
- **Adds real streamed-run regression tests** in `run.stream.test.ts` — a normal multi-turn streamed run (`currentTurn === 2` via two model requests) and the handled max-turn-0 boundary (`currentTurn === 0`). The prior test only mutated the private counter directly.
- **Switches the changeset to `fix:` format.**

`tsc` build clean and the full `agents-core` suite is green (158 files / 3397 tests). Ready for another look whenever you have a moment.

**Review by seratch — CHANGES_REQUESTED** [review]
Thanks for the update. This revision still needs one design change. Please keep currentTurn as the released writable field and set it from the streaming runner at the point a model turn is admitted: initialize it from resumed state, then update it only after the max-turn check and blocking input guardrails have passed, immediately before the model request starts. The getter changes the public field into a getter-only property and still reports 1 when a blocking first-turn input guardrail trips before any model request.

Please keep the existing normal and max-turn regressions and add blocking-guardrail and resumed-state coverage. This preserves the released property shape and makes currentTurn represent model turns actually admitted. Once those focused changes pass the current checks, this should be ready for another review.

File: packages/agents-core/src/result.ts [hunk 1]
```diff
@@ -309,9 +308,20 @@ export class StreamedRunResult<
...
+   * *before* it checks the `maxTurns` limit, so on a handled max-turn boundary
+   * (e.g. `maxTurns: 0`) the raw counter reads one higher than the number of
+   * turns actually admitted by the limit. Cap the public value at `maxTurns` so
+   * it always reflects the turns the limit allowed to run; when `maxTurns` is
+   * `null` (no limit) the raw counter is surfaced unchanged.
+   */
+  public get currentTurn(): number {
+    const current = this.state._currentTurn;
+    const max = this.state._maxTurns;
+    return max === null ? current : Math.min(current, max);
```
[the remark this answers is not in the document]
Comment (dkindlund) [reply packages/agents-core/src/result.ts]: Confirmed and fixed in 56a03d26 — thank you, this was a real bug and the diagnosis was exactly right.

`beginTurn` bumps `state._currentTurn` before both the `maxTurns` check and the blocking input guardrails, and the tripwire path never rolls it back, so surfacing the raw counter reported `1` for a guardrail-blocked first turn that never reached the model.

Taking the "derive from admitted model starts" option rather than rolling back the tripwire: `currentTurn` is now assigned in the streaming runner immediately after `prepareTurn` returns. Both the limit check and the blocking guardrails throw from inside `prepareTurn`, so reaching that assignment is precisely "this turn was admitted" — no rollback needed, and the `maxTurns: 0` boundary falls out of the same property.

Regression added in `run.stream.test.ts`: a blocking input guardrail (`runInParallel: false`) that trips before any model request now asserts `currentTurn === 0`. I mutation-tested it by assigning the raw counter before `prepareTurn` — the suite fails, so the test genuinely covers this case.

**Comment by dkindlund** [comment]
Thanks — reworked to the design you described in 56a03d26. Summary of what changed:

**`currentTurn` is a writable field again.** The getter is gone; the released property shape is preserved (`Object.getOwnPropertyDescriptor(...).writable === true`, no `get`). There is a test pinning that specifically, so it can't silently regress into a getter again.

**The streaming runner sets it at the admission point.** `#runStreamLoop` assigns `result.currentTurn = result.state._currentTurn` immediately after `prepareTurn` returns. That location does the work: both the `maxTurns` check and `runInputGuardrailsForTurn`'s blocking guardrails throw from *inside* `prepareTurn`, so reaching the assignment is exactly "this turn was admitted, and the model request is about to start."

A consequence worth flagging: **the cap is no longer needed.** `maxTurns: 0` and a first-turn guardrail tripwire both leave the value at `0` because the assignment is never reached, rather than being clamped after the fact. That also closes the blocking-guardrail case you and the Codex review both identified, which the cap did not cover.

**Seeded from resumed state** in the constructor, so a run continued from a serialized state does not restart its public count at `0`.

**Tests** — kept the normal multi-turn and handled max-turn regressions, and added:
- a blocking input guardrail (`runInParallel: false`) tripping before any model request → `currentTurn === 0`;
- a resumed streamed run (cancel mid-turn → serialize → `RunState.fromString` → resume streamed) carrying its turn count instead of restarting.

I also replaced the `result.test.ts` case you flagged. You were right that it only mutated the private counter — and under this design that assertion would have been actively misleading, since the field deliberately does *not* track later mutations of `RunState._currentTurn`. It is now three tests covering the released property shape, the resumed-state seed, and explicitly that later private-counter mutations are *not* reflected.

To check the tests actually cover the behaviour rather than merely passing, I mutation-tested them: removing the assignment, assigning the raw counter *before* `prepareTurn` (i.e. reproducing the reported bug), and removing the resumed-state seed each fail the suite.

**Changeset** rewritten to describe the new behaviour, in `fix: ...` form.

`packages/agents-core` is green (3399 passed). The only failures locally are two pre-existing `test/sandboxes/unixLocal.signals.test.ts` PTY cases that need a built `dist/`; they fail identically on the base commit with my changes stashed.

**Review by seratch — CHANGES_REQUESTED** [review]
Thanks for reworking this. Preserving `currentTurn` as a writable field and carrying resumed state now match the released API shape, but the assignment is still earlier than requested. In the current head, `currentTurn` advances before sandbox preparation, artifact and model-call preparation, the pending parallel-guardrail check, and the cancellation return. Any of those can stop the turn before `getStreamedResponseWithRetry` starts, leaving the result with an extra model turn that never occurred.

Please move the update to the actual model-request boundary, after all awaited preparation, guardrail, and cancellation checks and immediately before `sentInputToModel` and `getStreamedResponseWithRetry`. Add a parallel-guardrail regression asserting zero model calls and an unchanged `currentTurn`, and make the resumed-state test assert its exact expected count rather than `>= 1`. Once those focused changes pass, this should be ready for another review.

**Comment by seratch** [comment]
Thanks for working on this issue! Your contribution is included in https://github.com/openai/openai-agents-js/pull/1606
